sections/merge-accounts/burn/BurnActionBox.tsx

```diff
@@ -0,0 +1,333 @@
+	const { blockExplorerInstance } = Etherscan.useContainer();
+
+	const [gasPrice, setGasPrice] = useState<number>(0);
+	const [gasLimit, setGasLimitEstimate] = useState<number | null>(null);
+
+	const [error, setError] = useState<string | null>(null);
+	const [txModalOpen, setTxModalOpen] = useState<boolean>(false);
+	const [transactionState, setTransactionState] = useState<Transaction>(Transaction.PRESUBMIT);
+	const [buttonState, setButtonState] = useState<string | null>(null);
+	const [txHash, setTxHash] = useState<string | null>(null);
+	const txLink = useMemo(
+		() => (blockExplorerInstance && txHash ? blockExplorerInstance.txLink(txHash) : ''),
+		[blockExplorerInstance, txHash]
+	);
+
+	const onGoBack = () => router.replace(ROUTES.MergeAccounts.Home);
+
+	const {
+		targetCRatio,
+		SNXRate,
+		debtBalance,
+		issuableSynths,
+		currentCRatio,
+	} = useStakingCalculations();
+	const synthsBalancesQuery = useSynthsBalancesQuery(walletAddress);
+	const synthBalances =
+		synthsBalancesQuery.isSuccess && synthsBalancesQuery.data != null
+			? synthsBalancesQuery.data
+			: null;
+
+	const sUSDBalance = synthBalances?.balancesMap.sUSD
+		? synthBalances.balancesMap.sUSD.balance
+		: wei(0);
+
+	const stakeInfo = useCallback(
+		(burnAmount: Wei): Wei => wei(getStakingAmount(targetCRatio, burnAmount, SNXRate)),
+		[SNXRate, targetCRatio]
+	);
+
+	const burnAmount = debtBalance.gt(sUSDBalance) ? wei(sUSDBalance) : debtBalance;
+	const unstakeAmount = useMemo(() => {
+		const calculatedTargetBurn = Math.max(debtBalance.sub(issuableSynths).toNumber(), 0);
+		if (currentCRatio.gt(targetCRatio) && parseSafeWei(burnAmount, 0).lte(calculatedTargetBurn)) {
+			return stakeInfo(wei(0));
+		} else {
+			return stakeInfo(burnAmount);
+		}
+	}, [burnAmount, debtBalance, issuableSynths, targetCRatio, currentCRatio, stakeInfo]);
+
+	const getBurnTxData = useCallback(
+		(gas: Record<string, number>) => {
+			if (!(isAppReady && sourceAccountAddress)) return null;
+			const {
+				contracts: { Synthetix },
+			} = synthetixjs!;
+			return [Synthetix, 'burnSynths', [sourceAccountAddress, gas]];
+		},
+		[isAppReady, sourceAccountAddress, synthetixjs]
+	);
+
+	// gas
+
+	useEffect(() => {
+		let isMounted = true;
+		(async () => {
+			try {
+				setError(null);
+				const data: any[] | null = getBurnTxData({});
+				if (!data) return;
+				const [contract, method, args] = data;
+				const gasEstimate = await getGasEstimateForTransaction(args, contract.estimateGas[method]);
+				if (isMounted) setGasLimitEstimate(getNormalizedGasLimit(Number(gasEstimate)));
+			} catch (error) {
+				// console.error(error);
+				if (isMounted) setGasLimitEstimate(null);
+			}
+		})();
+		return () => {
+			isMounted = false;
+		};
+	}, [getBurnTxData, synthetixjs]);
+
+	// funcs
+
+	const connectOrBurn = async () => {
+		if (!isWalletConnected) {
+			return connectWallet();
+		}
+		burn();
+	};
+
+	const burn = async () => {
```

Comment:
would encourage you to use the new `useSynthetixTxn` mutative function in `useSynthetixQueries`--basically just put the data from `getBurnTxData` direcly into it, it will give you gas limit estimates and handle a lot of the state and erroring issues automatically for you

Reply:
done